internal/controllers/utils/conditions.go

```diff
@@ -111,10 +112,15 @@ func IsClusterProvisionPresent(cr *provisioningv1alpha1.ProvisioningRequest) boo
 	return condition != nil
 }
 
-// IsClusterProvisionCompleted checks if the cluster provision condition status is completed
+// IsClusterProvisionCompleted checks if the cluster provision condition status is completed.
+// The staleCondition is set when the ClusterDeployment's spec.installed has become true, but its status
+// conditions have not been properly updated due to the known issue (https://issues.redhat.com/browse/ACM-13064).
```

Comment:
Sure, everyone is currently on ACM 2.11. This can be removed after ACM 2.12 GA, but it doesn’t hurt to leave it for a while to allow people to transition to ACM 2.12.